src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -204,12 +316,13 @@ public int Capacity
                 {
                     throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_SmallCapacity);
                 }
+                Contract.EndContractBlock();
 
                 if (Capacity != value)
                 {
                     int newLen = value - m_ChunkOffset;
-                    var newArray = new char[newLen];
-                    Array.Copy(m_ChunkChars, 0, newArray, 0, m_ChunkLength);
+                    char[] newArray = new char[newLen];
+                    Array.Copy(m_ChunkChars, newArray, m_ChunkLength);
```

Comment:
The CoreRT version is better - it is faster to call `Array.Copy(m_ChunkChars, 0, newArray, 0, m_ChunkLength)`